### DISCUSSION TITLE
Change pydantic default error message in fastapi?

### DISCUSSION THREAD

**Discussion by binaryantcom** [fastapi/fastapi#9420] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [ ] I commit to help with one of those options 👆

### Example Code

```python
import uvicorn
from fastapi import FastAPI, Query

app = FastAPI()
 
@app.get("/")
async def read_items(name:str= Query(max_length=5,error_message='name too long'),
                     age:int= Query(error_message='age need intger')):
   return '...'
 
if __name__ == '__main__':
    uvicorn.run(app="main:app", host="localhost", port=5000)
```

### Description

I type:http://localhost:5000/?name=asdfasdfasdfasdreerer&age=s and the error message is below:
{"detail":[{"loc":["query","name"],"msg":"ensure this value has at most 5 characters","type":"value_error.any_str.max_length","ctx":{"limit_value":5}},{"loc":["query","age"],"msg":"value is not a valid integer","type":"type_error.integer"}]}

**I want to show my own error message**
{"detail":[{"loc":["query","name"],"msg":"**name too long**","type":"value_error.any_str.max_length","ctx":{"limit_value":5}},{"loc":["query","age"],"msg":"**age need integer**","type":"type_error.integer"}]}

**'error message' is a assumed property** 

### Operating System

Windows

### FastAPI Version

0.95.0

### Python Version

3.10.9

**Comment by jgould22** [comment]
Your question is not quite clear however one way to do this is to use pythons[ gettext module](https://docs.python.org/3/library/gettext.html) and use the `Accept-Language` header to determine which locale the client would prefer.

**Reply by binaryantcom** [reply]
The content is updated.Did I make it clear?

**Comment by yinziyan1206** [comment]
You could change the `error_msg_templates` in pydantic model config.

**Comment by YuriiMotov** [comment, marked as answer]
I don't think such feature would ever been implemented - it seems not good to me to specify one error message for all possible errors for this parameter.

But you can implement such logic using the following approach:

```py
from fastapi import FastAPI, Query, Request
from fastapi.exceptions import RequestValidationError
from fastapi.responses import JSONResponse

app = FastAPI()


def format_error_msg(err: dict):
    if err["type"] == "string_too_long":
        err["msg"] = f"{err['loc'][-1]} too long"

    if err["type"] == "int_parsing":
        err["msg"] = f"{err['loc'][-1]} need integer"

    return err


@app.exception_handler(RequestValidationError)
async def validation_exception_handler(request: Request, exc: RequestValidationError):
    errors = [format_error_msg(err) for err in exc.errors()]
    return JSONResponse(
        status_code=422,
        content={"detail": errors},
    )


@app.get("/")
async def read_items(
    name: str = Query(max_length=5),
    age: int = Query(),
):
    return "..."

```

```
curl 'http://127.0.0.1:8000?name=loooong_name&age=not_int'

{
    "detail": [
        {
            "type": "string_too_long",
            "loc": ["query", "name"],
            "msg": "name too long",
            "input": "loooong_name",
            "ctx": {"max_length": 5},
        },
        {
            "type": "int_parsing",
            "loc": ["query", "age"],
            "msg": "age need integer",
            "input": "not_int",
        },
    ]
}

```
